File: src/checkers-api/Models/Responses/ValidMove.cs
using checkers_api.Models.GameModels;

namespace checkers_api.Models.Responses;

public class ValidMove
{
    public Location Destination { get; }
    public IEnumerable<Move> MoveSequence { get; }

    public ValidMove(Location destination, IEnumerable<Move> moveSequence)
    {
        Destination = destination;
        MoveSequence = moveSequence;
    }
}

File: src/checkers-client/src/models/game/validMove.ts
import Location from "./location";
import Move from "./move";

export default class ValidMove {
    public readonly destination: Location;
    public readonly moveSequence: Move[];

    constructor (destination: Location, moveSequence: Move[]) {
        this.destination = destination;
        this.moveSequence = moveSequence;
    }
}